### ISSUE TITLE
YOLOv11 torch和trt精度对不齐

### ISSUE THREAD

**Issue by lyc728** [ultralytics/ultralytics#17676]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

YOLOv11 精度损失

### Environment

Version: 1.13.1+cu116
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /usr/local/lib64/python3.9/site-packages
Requires: typing-extensions
Required-by: torchvision, torchtext, torchaudio, ultralytics, ultralytics-thop, torchmetrics, timm, pytorch-lightning, OpenNMT-py

 TensorRT.trtexec [TensorRT v8401] # trtexec

### Minimal Reproducible Example

torch 代码
`from ultralytics import YOLO

model = YOLO("/data/yolov_11/runs/train/exp11/weights/best.pt", task="detect")
path = model.export(format="onnx", simplify=True, device=0, opset=12, dynamic=False, imgsz=640)`

trt代码
`polygraphy run best.onnx --save-engine=best.plan --onnxrt --trt --workspace $((1<<32)) --atol 1e-3 --rtol 1e-5 --verbose`

### Additional

日志里面显示精度过不了 同时推理发现确实有问题  trt推理代码参考
https://github.com/emptysoal/TensorRT-YOLO11/tree/main/python/detect

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lyc728, thank you for reaching out to Ultralytics 🚀! We appreciate your initiative in trying to help advance our codebase. This is an automated response and an Ultralytics engineer will be with you shortly to further assist!

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us thoroughly investigate and debug the issue.

In the meantime, you could explore our [Docs](https://docs.ultralytics.com) for additional insights on common issues and solutions. We have detailed sections for [Python Usage](https://docs.ultralytics.com/usage/python/) and [CLI Usage](https://docs.ultralytics.com/usage/cli/) which might be helpful.

To further troubleshoot, ensure you're using the latest version of the Ultralytics package. You can upgrade by running the following command:

```bash
pip install -U ultralytics
```

YOLO models can be run in various up-to-date verified environments. Check one of these if you're working with different setups:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM or **Amazon** Deep Learning AMI. Check our quickstart guides for [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For real-time assistance, feel free to join our [Discord community](https://ultralytics.com/discord) 🎧. If you prefer more in-depth discussions or sharing your experiences with others, visit our [Discourse forum](https://community.ultralytics.com) or join the conversation on our [Subreddit](https://reddit.com/r/ultralytics).

🛠️ Meanwhile, our engineers will dive into this and get back to you soon. Thank you for your patience and collaboration!

**Comment by glenn-jocher** [comment]
@lyc728 thank you for your detailed report. It seems like you're experiencing precision discrepancies between PyTorch and TensorRT for YOLOv11. This can often occur due to differences in how operations are handled between frameworks. Ensure your ONNX export uses compatible opsets and double-check your TensorRT configuration for any precision settings. You might also want to review [this guide](https://docs.ultralytics.com/integrations/tensorrt/) for further insights into optimizing TensorRT precision. If you continue to face issues, feel free to share your findings or submit a PR with any solutions you discover.
